Core/DetectedTracker.swift

```diff
@@ -27,12 +27,14 @@ public struct DetectedTracker {
     public let knownTracker: KnownTracker?
     public let entity: Entity?
     public let blocked: Bool
+    public let pageUrl: String?
     
-    public init(url: String, knownTracker: KnownTracker?, entity: Entity?, blocked: Bool) {
+    public init(url: String, knownTracker: KnownTracker?, entity: Entity?, blocked: Bool, pageUrl: String? = nil) {
```

Comment:
Shouldn't pageURL be non-optional?

Reply:
ContentBlockerUserScript (i.e. the other user script) also uses DetectedTracker but doesn't receive the page URL from the JS script, maybe it should? I don't 100% know what is done as a result of any message from that script.

I can see SiteRating tracks installed surrogates for SiteRating reports, I guess they probably suffer from the same race condition so should have the same check.

Reply:
Yes, I think it should - otherwise `guard tracker.pageUrl == url.absoluteString else { return }` won't let the code through as pageURL is nil - it happens if you break on that return and visit page with surrogates.

Reply:
I've added the page url to the messages sent from the other script and made pageURL not optional now